src/map.rs

```diff
@@ -65,6 +65,27 @@ pub struct Map {
     pub user_type: Option<String>,
 }
 
+impl fmt::Debug for Map {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Map")
+            .field("version", &self.version)
+            .field("orientation", &self.orientation)
+            .field("width", &self.width)
+            .field("height", &self.height)
+            .field("tile_width", &self.tile_width)
+            .field("tile_height", &self.tile_height)
+            .field("stagger_axis", &self.stagger_axis)
+            .field("stagger_index", &self.stagger_index)
+            .field("tilesets", &format!("{} tilesets", self.tilesets.len()))
+            .field("layers", &format!("{} layers", self.layers.len()))
```

Comment:
Instead of debug-printing all tilesets and layers, we only debug-print the length of `self.tilesets` and `self.layers`.

I don't know if this is a viable approach as it calls `len` and allocates `String` with `format!`, which would be not as performant as desired. On the other hand, we're talking about `Debug` here, which should/must be only relied upon debug builds, not release builds.